up-l1/zenoh.adoc

```diff
@@ -98,22 +108,24 @@ Different message types **MUST** use different Zenoh API.
 
 === URI Mapping
 
-UUri **MUST** be able to be serialized into link:../basics/uri.adoc#42-micro-uris[micro form].
-Then we can transform the micro-form UUri, which is `u8` array, into string in lowercase hexadecimal representation.
+UUri **MUST** be able to be serialized into link:../basics/uri.adoc#short-uris[short form].
+Then we can transform the short-form UUri into string for Zenoh key.
 
 Zenoh key for uProtocol consists of three parts and they are separated by `/`.
-Zenoh key **MUST** start with `upl/` or `upr/`.
-While the local UUri maps to `upl/`, the remote UUri maps to `upr/`.
-If UAuthority is defined in UUri, it **MUST** be put in the second part.
-Then the rest of the UUri **MUST** be put in the last part.
 
-To be more specific:
+* `up/`: Zenoh key **MUST** always start with `up/` and the scheme (`up://`) in the short UUri is always removed.
+* UAuthority: Although UAuthority does not exist in local UUri, it **MUST** be used in Zenoh key. We can use the UAuthority which is initialized in the constructor in this case.
+* `UE_ID`, `UE_VER_MAJOR`, and `URESOURCE_ID`: These should be kept in Zenoh Key and also separated by `/`.
+
+To be more specific, the format of Zenoh Key is:
+
+`up/[UAuthoriy]/[UE_ID]/[UE_VER_MAJOR]/[URESOURCE_ID]`
 
-* The local UUri without UAuthority: `upl/[hexadecimal representation of the UUri]`
-* The remote UUri with UAuthority: `upr/[hexadecimal representation of the UAuthority]/[hexadecimal representation of the rest of the UUri]`
+The UAuthority of short-form UUri is case-insensitive, but Zenoh key is case-sensitive.
+We **MUST** transform all the alphabets of the UAuthority into lowercase.
```

Comment:
hum.... perhaps the serializer should do this in lieu of inside of the transport and we make short form URI case sensitive, what say you @sophokles73 & @AnotherDaniel ?

Reply:
So far, we have tried to keep very close to the URI definition of RFC3986. As I pointed out to @evshary in https://github.com/eclipse-uprotocol/up-spec/pull/121#issuecomment-2078684079, we should be able to make small restrictions while maintaining overall compatibility. Making authority case-sensitive would potentially start breaking compatibility IMHO. I do not see an issue with letting the transport decide what to do with the authority in order to make it work for them. Restricting the allowed characters in the authority name might still be worthwhile, though, in order to increase general _interoperability_ with transports.